plugins/wporg-learn/inc/class-workshop.php

```diff
@@ -0,0 +1,146 @@
+<?php
```

Comment:
Broadly speaking I think you're using custom post types and taxonomies just fine. It seems like an appropriate way to store everything.